### DISCUSSION TITLE
How to setup Shiny.Notifications on net7.0-android & net7.0-ios?

### DISCUSSION THREAD

**Discussion by vallgrenerik** [shinyorg/shiny#1194] [Q&A]
Hello!  😊

I'm new to Shiny and starting to investigate how to use it.  
I really think it looks promising.   
I'm in the process of porting a old Xamarin.Android and Xamarin.iOS application to .net7.0.  

When I read the documentation and try the samples I guess that plain net7.0-android and net7.0-ios is not the main focus of this project. The examples are for MAUI and I completely understand that.  

But are there any technical limitations for using the new (still alpha versions) Shiny.Notifications on net7.0?   
I've done some testing and the notifications partly work. 
It seems to work good on iOS and partly on Android. 

This is a example from MainActivity: (the project is a file new net7.0-android) 
```cs
[Activity(Label = "@string/app_name", MainLauncher = true)]
public class MainActivity : Activity
{
    private IHost _build;

    protected override async void OnCreate(Bundle? savedInstanceState)
    {
        base.OnCreate(savedInstanceState);
        
        this.SetContentView(Resource.Layout.activity_main);
        var builder = new HostBuilder();
        builder.Services.AddNotifications();
        this._build = builder.Build();


        if (ContextCompat.CheckSelfPermission(this, Manifest.Permission.PostNotifications) != Permission.Granted)
        {
            ActivityCompat.RequestPermissions(this, new[] { Manifest.Permission.PostNotifications }, 0);
        }

        var findViewById = this.FindViewById<Button>(Resource.Id.MyButton);
        findViewById.Click += async (s, e) => { await this.FindViewByIdOnClick(s, e); };
    }

    protected async Task FindViewByIdOnClick(object? sender, EventArgs e)
    {        
        var notificationManager = this._build.Services.GetService<INotificationManager>();
        await notificationManager.Send("Hello", "Testing");
    }
}
```

The notification appears as expected. 

But when I try to schedule the notification, like this: 

``` cs
protected async Task FindViewByIdOnClick(object? sender, EventArgs e)
    {        
        var notificationManager = this._build.Services.GetService<INotificationManager>();
        await notificationManager.Send("Hello", "Testing", null, DateTime.Now.AddSeconds(5));
    }
```
The notification does not appear. 

When I'm using the source code from this repo I can see that an exception comes from ```ShinyBroadcastReceiver.cs``` when calling: 
    protected T Resolve<T>() => Host.Current.Services.GetRequiredService<T>()!;  
The exception message is: ```ServiceProvider is not initialized - This means you have not setup Shiny correctly```  

So I guess I have to try to Initilize in some other way.  
When I try it on my production app, using MvvmCross I get the same error. 

Do you have any ideas how to solve this, or if its possible?

Thanks alot if you have the time to answer ❤️

**Comment by aritchie** [comment]
If you are going through all of this just for notifications, I wouldn't recommend it.  Please also note that I'm not supporting anything around the alphas.  I'll help you with this set of answers below and that's it!

1. Initializing shiny in any form in the mainactivity is not supported.  You have to initialize at the android application level.
2. You have to call .Run() on the host after you have built it.  Follow the hosting models that are currently in the master branch.
3. You will need to call all of the shiny hooks manually as shown in the hosting models.
